src/components/Widget/index.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { PROP_TYPES } from 'constants';
+import { toggleChat, addUserMessage } from 'actions';
+
+import WidgetLayout from './layout';
+
+class Widget extends Component {
+  toggleConversation = () => {
+    this.props.dispatch(toggleChat());
+  }
+
+  handleMessageSubmit = (event) => {
+    event.preventDefault();
+    const userInput = event.target.message.value;
+    if (userInput) {
+      this.props.dispatch(addUserMessage(userInput));
+      this.props.handleNewUserMessage(userInput);
+    }
+  }
+
+  render() {
+    return (
+      <WidgetLayout
+        toggleConversation={this.toggleConversation}
+        sendMessage={this.handleMessageSubmit}
+        title={this.props.title}
+        subtitle={this.props.subtitle}
+        senderPlaceHolder={this.props.senderPlaceHolder}
+        stylesInjected={this.props.stylesInjected}
+        profileAvatar={this.props.profileAvatar}
+      />
+    );
+  }
+}
+
+Widget.propTypes = {
+  title: PropTypes.string,
+  subtitle: PropTypes.string,
+  handleNewUserMessage: PropTypes.func.isRequired,
+  senderPlaceHolder: PropTypes.string,
+  stylesInjected: PROP_TYPES.STYLES,
```

Comment:
Since this is the object that in a way contains all the styles that are going to be injected, it has a propstyle setting a shape. The others are just objects with random styles.